pkg/operator/statefulset.go

```diff
@@ -24,74 +24,154 @@ import (
 
 const configReloaderPort = 8080
 
+type ConfigReloader struct {
+	name           string
+	config         ReloaderConfig
+	reloadURL      url.URL
+	listenLocal    bool
+	localHost      string
+	logFormat      string
+	logLevel       string
+	additionalArgs []string
+	volumeMounts   []v1.VolumeMount
+	shard          int32
+}
+
+type option = func(*ConfigReloader)
+
+func ReloaderRunOnce() option {
+	return func(c *ConfigReloader) {
+		c.additionalArgs = append(c.additionalArgs, fmt.Sprintf("--watch-interval=%d", 0))
+	}
+}
+
+func ListenAddress() option {
+	return func(c *ConfigReloader) {
+		var listenAddress string
+		if c.listenLocal {
+			listenAddress = c.localHost
+		}
+		c.additionalArgs = append(c.additionalArgs, fmt.Sprintf("--listen-address=%s:%d", listenAddress, configReloaderPort))
+	}
+}
+
+func ReloadURL() option {
+	return func(c *ConfigReloader) {
+		c.additionalArgs = append(c.additionalArgs, fmt.Sprintf("--reload-url=%s", c.reloadURL.String()))
+	}
+}
+
+func ReloaderResources(rc ReloaderConfig) option {
+	return func(c *ConfigReloader) {
+		c.config = rc
+	}
+}
+
+func ReloaderURL(reloadURL url.URL) option {
+	return func(c *ConfigReloader) {
+		c.reloadURL = reloadURL
+	}
+}
+
+func ListenLocal(listenLocal bool) option {
+	return func(c *ConfigReloader) {
+		c.listenLocal = listenLocal
+	}
+}
+
+func LocalHost(localHost string) option {
+	return func(c *ConfigReloader) {
+		c.localHost = localHost
+	}
+}
```

Comment:
It might make sense to keep only one function and move the logic to the caller

```suggestion
func ListenLocal(localhost string) option {
	return func(c *ConfigReloader) {
		c.localHost = localHost
	}
}
```

Then in pkg/prometheus/statefulset.go

```go
if p.Spec.ListenLocal {
  reloaderOpts = append(reloaderOpts, operator.ListenLocal(c.Localhost))
}
```